projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -127,14 +140,14 @@ def to_css(self) -> str:
 def include_raw(ctx, name) -> Markup:  # noqa: ANN001
     """Normal jinja2 {% include %} doesn't escape {{...}} which appear in React's source code"""
     env = ctx.environment
-    # Escape </script> to prevent 3rd party JS terminating the local report bundle.
+    # Escape </script> to prevent 3rd party JS terminating the local app bundle.
     # Note there's an extra "\" because it needs to be escaped at both the python and JS level
     src = env.loader.get_source(env, name)[0].replace("</script>", r"<\\/script>")
     return Markup(src)
 
 
-class ReportFileWriter:
-    """Provides shared logic for saved and served local report writers"""
+class AppFileWriter:
```

Comment:
is this needed as a standalone class or can it be moved into the Processor base?

Reply:
Yeah it makes sense to move the writer logic into the `Processor` hierarchy. I've added it to an intermediate `LocalProcessor` as only the `Saver` and `Server` classes need to use it.